PWGHF/D2H/Tasks/taskFlowCharmHadrons.cxx

```diff
@@ -0,0 +1,339 @@
+        cent = collision.centFV0A();
+        break;
+      case 1: // T0M
+        cent = collision.centFT0M();
+        break;
+      case 2: // T0A
+        cent = collision.centFT0A();
+        break;
+      case 3: // T0C
+        cent = collision.centFT0C();
+        break;
+      default:
+        break;
+    }
+
+    return cent;
+  }
+
+  /// Compute the scalar product
+  /// \param collision is the collision with the Q vector information and event plane
+  /// \param candidates are the selected candidates
+  template <int decayChannel, typename T1>
+  void runFlowAnalysis(CollsWithQvecs::iterator const& collision,
+                       T1 const& candidates)
+  {
+    float QvecX = -999.;
+    float QvecY = -999.;
+    float amplQvec = -999.;
+    float evtPl = -999.;
+    switch (qvecDetector.value) {
+      case 0: // FT0c
+        QvecX = collision.qvecFT0CRe();
+        QvecY = collision.qvecFT0CIm();
+        break;
+      case 1: // FT0a
+        QvecX = collision.qvecFT0ARe();
+        QvecY = collision.qvecFT0AIm();
+        break;
+      case 2: // FT0m
+        QvecX = collision.qvecFT0MRe();
+        QvecY = collision.qvecFT0MIm();
+        break;
+      case 3: // FV0a
+        QvecX = collision.qvecFV0ARe();
+        QvecY = collision.qvecFV0AIm();
+        break;
+      case 4: // TPC Positive
+        QvecX = collision.qvecBPosRe();
+        QvecY = collision.qvecBPosIm();
+        amplQvec = collision.nTrkBPos();
+        break;
+      case 5: // TPC Negative
+        QvecX = collision.qvecBNegRe();
+        QvecY = collision.qvecBNegIm();
+        amplQvec = collision.nTrkBNeg();
+        break;
+      default:
+        break;
+    }
+    evtPl = epHelper.GetEventPlane(QvecX, QvecY);
+
+    float cent = GetCent(collision, centDetector.value);
+
+    for (auto const& cand : candidates) {
+      float massCand = 0.;
+      float outputMl = -999.;
+
+      if constexpr (std::is_same<T1, CandDsData>::value) {
+        switch (decayChannel) {
+          case decayChannel::DsToKKPi:
+            massCand = hfHelper.invMassDsToKKPi(cand);
+            if (storeMl)
+              outputMl = cand.mlProbDsToPiKK()[classMl.value];
+            break;
+          case decayChannel::DsToPiKK:
+            massCand = hfHelper.invMassDsToPiKK(cand);
+            if (storeMl)
+              outputMl = cand.mlProbDsToKKPi()[classMl.value];
+            break;
+          default:
+            break;
+        }
+      } else if constexpr (std::is_same<T1, CandDplusData>::value) {
+        massCand = hfHelper.invMassDplusToPiKPi(cand);
+        if (storeMl)
+          outputMl = cand.mlProbDplusToPiKPi()[classMl.value];
+      }
+      float ptCand = cand.pt();
+      float phiCand = cand.phi();
+
+      // TPC only
+      if (qvecDetector.value == 4 || qvecDetector.value == 5) {
+        float ampl = amplQvec - 3.;
+        std::vector<float> tracksQx = {-999., -999., -999.};
+        std::vector<float> tracksQy = {-999., -999., -999.};
```

Comment:
```suggestion
        std::vector<float> tracksQx = {};
        std::vector<float> tracksQy = {};
```
Since in `GetQvecDtracks` you push back the values you do not overwrite your initialisation, thus in the following for loop you are always getting `-999.`. 
It would be better to use `std:arrays` since you know the size you need.